### ISSUE TITLE
Style/GuardClause does not trigger on longer (but not too long) line

### ISSUE THREAD

**Issue by JeanMertz** [bbatsov/rubocop#3209]
```
$ cat .rubocop.yml
AllCops:
  TargetRubyVersion: 2.3
```

```
$ rubocop -V
0.40.0 (using Parser 2.3.1.0, running on ruby 2.3.0 x86_64-darwin15)
```

```
$ cat test.rb 
# frozen_string_literal: true

def hello_world
  if hello
    return foo if bar
    return baz unless box
  end
end
```

```
$ rubocop test.rb 
Inspecting 1 file
C

Offenses:

test.rb:4:3: C: Use a guard clause instead of wrapping the code inside a conditional expression.
  if hello
  ^^

1 file inspected, 1 offense detected
```

```
$ sed -ie 's/bar/bar_is_now_a_bit_longer_than_before/' test.rb
```

```
$ rubocop test.rb 
Inspecting 1 file
.

1 file inspected, no offenses detected
```

A couple of questions:

1. why is this non-guard setup suddenly acceptable? Even when converted to a guard statement, the line would still be within the original limit of 80 characters.
2. even if the line length is exceeded, is that really something this cop should concern itself with? There are other ways to solve that (extracting long logics into their own method, or, if all else fails, disabling the cop in this specific case)

**Comment by jonas054** [comment]
That's a bug. There's no reason why making that line slightly longer should make any difference.

**Comment by Drenmi** [comment]
This seems to go deeper than what is immediately apparent. I am fiddling around with it now, and it has caused **a lot** of false negatives in RoboCop itself. I might have a PR incoming soon.
